pkg/scheduler/cache/cache.go

```diff
@@ -1225,8 +1226,21 @@ func (sc *SchedulerCache) GetMetricsData() {
 				if len(warnings) > 0 {
 					klog.V(3).Infof("Warning querying Prometheus: %v", warnings)
 				}
+				if res == nil || res.String() == "" {
+					klog.V(4).Infof("Warning querying Prometheus: no data found for %s", queryStr)
+					continue
+				}
 
-				rowValues := strings.Split(strings.TrimSpace(res.String()), "=>")
+				klog.V(4).Infof("Query prometheus res %s", res.String())
+				switch res.Type() {
```

Comment:
So only `pmodel.ValVector` is necessary,  if it is another type, just skip the metric process? If that, how about use a `if` to judge that?

Reply:
Another issue I doubt is that if something goes wrong and Volcano cannot get the target metrcs, would it  process the following logic with pod's request resource? Namely, swtich to the default mode.

Reply:
in case [prometheus.rules](https://github.com/volcano-sh/volcano/blob/master/docs/design/usage-based-scheduling.md#prometheus-rule-configuration), prometheus.api.query only type pmodel.ValVector can occurred, so ignore anyother type
plugin usage and predicates can process parallelly, they shouldnt touch each other, no situation "switch" is expected to happen, can imports both as backups for each other